src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java

```diff
@@ -56,10 +59,28 @@ public void deleteAll(@NonNull UUID networkUuid) {
     @Override
     public List<EquipmentInfos> search(@NonNull final String query) {
         NativeSearchQuery nativeSearchQuery = new NativeSearchQueryBuilder()
-                .withQuery(QueryBuilders.queryStringQuery(query))
-                .withPageable(PageRequest.of(0, PAGE_MAX_SIZE))
-                .build();
+            .withQuery(QueryBuilders.queryStringQuery(query))
+            .withPageable(PageRequest.of(0, PAGE_MAX_SIZE))
+            .build();
+
         SearchHits<EquipmentInfos> searchHits = elasticsearchOperations.search(nativeSearchQuery, EquipmentInfos.class);
-        return searchHits.stream().map(SearchHit::getContent).collect(Collectors.toList());
+
+        List<EquipmentInfos> infos = searchHits
+            .stream()
+            .map(SearchHit::getContent)
+            .collect(Collectors.toList());
+
+        return Stream.of(
+                infos.stream().filter(e -> e.getType().equals(EquipmentType.SUBSTATION.name())).sorted(Comparator.comparing(EquipmentInfos::getId)),
+                infos.stream().filter(e -> e.getType().equals(EquipmentType.VOLTAGE_LEVEL.name())).sorted(Comparator.comparing(EquipmentInfos::getId)),
+                infos.stream().filter(e -> e.getType().equals(EquipmentType.LINE.name())).sorted(Comparator.comparing(EquipmentInfos::getId)),
+                infos.stream().filter(e -> e.getType().equals(EquipmentType.TWO_WINDINGS_TRANSFORMER.name())).sorted(Comparator.comparing(EquipmentInfos::getId)),
+                infos.stream().filter(e -> e.getType().equals(EquipmentType.THREE_WINDINGS_TRANSFORMER.name())).sorted(Comparator.comparing(EquipmentInfos::getId)),
+                infos.stream().filter(e -> e.getType().equals(EquipmentType.HVDC.name())).sorted(Comparator.comparing(EquipmentInfos::getId)),
+                infos.stream().filter(e -> EquipmentType.isInjectionType(e.getType())).sorted(Comparator.comparing(EquipmentInfos::getId)),
+                infos.stream().filter(e -> EquipmentType.isSwitchType(e.getType())).sorted(Comparator.comparing(EquipmentInfos::getId))
```

Comment:
Maybe splitting first the 'infos' stream into a Map<EquipmentType, List<EquipmentInfos>> with Collectors.groupingBy(EquipmentInfos::getType)
and the sort each List in the map
would be more efficient ?